classes/auth.php

```diff
@@ -100,10 +100,13 @@ public function __construct() {
         $this->defaults['idpdefaultname'] = get_string('idpnamedefault', 'auth_saml2');
         $this->defaults['flagmessage'] = get_string('flagmessage_default', 'auth_saml2');
         $this->authtype = 'saml2';
-        $mdl = new moodle_url($CFG->wwwroot);
+
+        $baseurl = optional_param('baseurl', $CFG->wwwroot, PARAM_URL);
```

Comment:
Could this have the potential to overwrite the current prod cert via using the default value?

Reply:
Nope, if the base != wwwroot then it's an independent file

Reply:
NVM this param only ever concerns the metadata XML, not the underlying certs